reactioncommerce/reaction
# Part 2: Filter by CSV

Resolves #5362 
Impact: **minor**  
Type: **feature**

## Issue
#5362 (second half)

## Solution

1. All uses of dropzone in reaction are upgraded to v10
(Note that in TagForm, this was accomplished by adding a private component; I was literally unable to make it work any other way, and I certainly tried many ways)

2. Multiple CSV files can be selected with the Import button, which itself uses dropzone

3. Catalyst Chip components then show the names of the files that were uploaded. The Chip components are deletable; Deleting the chips will remove them from the import process.

4. Pressing the 'Filter products' button then causes the CSV files to be parsed into values.
4a. A designsystem (not Catalyst) InlineAlert is shown, saying that the product list has been filtered, and giving the count of filtered items. It is closable.
4b. The Card containing filtering UX is removed.
4c. Catalyst Chip components containing the names of files uploaded, appear below the Actions dropdown. They are deletable, and deleting them causes the list to be re-filtered without the contents of those files.
(Note that accomplishing the passing down of files to the Products listing and handling the associated delete actions involved a few instances of passing props down several levels)

## Breaking changes
None


## Testing
1. Verify that the steps listed in 'Solution' are happening as described
2. Verify that uploading via react-dropzone wasn't broken where it was updated to v10:
2a. Uploading Tag Hero images
2b. Uploading Store images
2c. Uploading Product media
